xapian-applications/omega/postprocess.py

```diff
@@ -0,0 +1,182 @@
+# Postprocess click data files.
+#
+# Copyright (C) 2017 Vivek Pal
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301
+# USA
+
+"""Postprocesses search and clicks data files.
+
+Generates the final clickstream log file and Query file for Xapian Letor
+module from that log file.
+"""
+
+import argparse
+import collections
+import csv
+import sys
+
+parser = argparse.ArgumentParser(
```

Comment:
This should be run only if `__name__ == "__main__"`, ie if this is called as a script. If it's imported as a module, we don't want to do this (which will allow you to write tests for the functions).